### PR TITLE
Use Travis Caching to Speed up OpenBLAS

### PR DESCRIPTION
**Pull request by jonlawlor**
Travis has a caching feature, where you can "persist" a directory across
CI calls, detailed at https://docs.travis-ci.com/user/caching/

This change creates a cache in the OpenBLAS.cache/ directory (only
available on travis) which holds the OpenBLAS and netlib libraries and
headers used.  The installation script checks to see if the cache
contains the latest version of OpenBLAS by checking the commit id, and
if there is a newer commit on the OpenBLAS git repo, then the cache is
cleared and rebuilt.  There is no handling of the netlib cblas
versioning at the moment.

Each item on the built matrix gets a different cache, so we will still
build OpenBLAS at least as many times as we have go versions in the
build matrix, but after that they won't build again until OpenBLAS
updates.

### CODE REVIEW

**Comment by jonlawlor** [comment]
Something about this is tripping up the osx builds...

**Comment by jonlawlor** [comment]
Works now.  This supersedes #174 because this approach is simpler and does not require coordination between a docker hub repo and a github repo.  It appears to make OpenBLAS builds take roughly the same amount of time as the native BLAS builds.

PTAL

File: .travis/linux/ATLAS/test.sh [hunk 1]
```diff
@@ -5,3 +5,5 @@ go get -d -t -v ./...
 go test -a -v ./...
 go test -a -tags noasm -v ./...
 if [[ $TRAVIS_SECURE_ENV_VARS = "true" ]]; then bash -c "$GOPATH/src/github.com/$TRAVIS_REPO_SLUG/.travis/test-coverage.sh"; fi
+
+set +ex
```
Comment (kortschak) [inline .travis/linux/ATLAS/test.sh]: Can you include a comment about this in the scripts (also in the lapack PR).

**Comment by kortschak** [comment]
Please add final crlf to files missing them.

**Comment by jonlawlor** [comment]
Done, waiting on changes needed for gonum/lapack#156.

**Comment by jonlawlor** [comment]
This has the same changes as gonum/lapack#156, can I get a LGTM?

**Comment by vladimir-ch** [comment]
LGTM

**Pull request merged**